ngraph/core/src/op/log_softmax.cpp

```diff
@@ -0,0 +1,94 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/log_softmax.hpp"
+
+#include "itt.hpp"
+#include "ngraph/runtime/reference/log_softmax.hpp"
+#include "ngraph/util.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v5::LogSoftmax, "LogSoftmax", 5);
+
+op::v5::LogSoftmax::LogSoftmax(const Output<Node>& arg, const size_t axis)
+    : Op({arg})
+    , m_axis(axis)
+{
+    constructor_validate_and_infer_types();
+}
+
+bool op::v5::LogSoftmax::visit_attributes(AttributeVisitor& visitor)
+{
+    visitor.on_attribute("axis", m_axis);
+    return true;
+}
+
+void op::v5::LogSoftmax::validate_and_infer_types()
+{
+    const PartialShape& input_shape = get_input_partial_shape(0);
+    if (input_shape.rank().is_static())
+        NODE_VALIDATION_CHECK(this,
+                              m_axis < input_shape.rank().get_length() &&
+                                  m_axis >= -input_shape.rank().get_length(),
```

Comment:
This check fails if the axis value is different than the default(zero). https://ideone.com/G7J0ro the second part of the condition is done in the "unsigned mode" meaning that the negated rank value gets casted to `size_t` and overflows making the comparison always false even with correct values of the axis attribute.